sa/database_test.go

```diff
@@ -188,8 +208,88 @@ func TestAdjustMySQLConfig(t *testing.T) {
 			"max_statement_time": "0",
 		},
 	}
-	adjustMySQLConfig(conf)
+	err = adjustMySQLConfig(conf)
+	test.AssertNotError(t, err, "unexpected err setting server variables")
 	test.AssertDeepEquals(t, conf.Params, map[string]string{
 		"sql_mode": "'STRICT_ALL_TABLES'",
 	})
+
+	conf = &mysql.Config{
```

Comment:
especially now that you've split the code itself out into a separate file, I'd move these test cases to //sa/sysvars_test.go, and directly call checkMariaDBSystemVariables rather than calling adjustMySQLConfig.

Once you do that, I think it would make a lot of sense to make these a "table test". Something like:

```go
func TestCheckMariaDBSystemVariables(t *testing.T) {
  type testCase struct {
    name string
    key string
    value string
    expectErr string
  }

  for _, tc := range []testCase{
    { test case goes here },
    { next test case goes here },
  } {
    t.Run(tc.name, func() {
      err := checkMariaDBSystemVariables(tc.key, tc.value)
      if tc.expectErr == "" {
        test.AssertNotError(t, err)
      } else {
        test.AssertContains(t, err.Error(), tc.expectErr)
      }
    }
  }
}
```

or something very approximately like that (forgive the sloppiness, github comments aren't the best IDE :P)